### PR TITLE
Creates the TapAndHold component

### PR DESCRIPTION
**Pull request by normancapule**
## Description
Creates a Tap and Hold button to be used by the OnsiteDetailsScreen in Apollo.

## Ticket
https://app.clubhouse.io/secondcloset/story/19159/apollo-create-tap-and-hold-button-component

## Screenshots
flow

### CODE REVIEW

**Review by alexcheng068 — APPROVED** [review]

**Review by sephlao — APPROVED** [review]

File: src/components/TapAndHoldButton/TapAndHoldButton.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  const onHoldText = "Release to Cancel";
+
+  const onCompletedState = progressValue === 1; // the bar has fully loaded
+
+  if (loading) {
+    return (
+      <S.Container>
+        <S.ClickButton disabled={true}>
+          <ActivityIndicator size={"small"} color={colors.GREY_MID} />
+        </S.ClickButton>
```
Comment (rachelyellow) [inline src/components/TapAndHoldButton/TapAndHoldButton.tsx:39]:
```
        <S.ClickButton disabled>
          <ActivityIndicator size="small" color={colors.GREY_MID} />
        </S.ClickButton>
```

File: src/components/TapAndHoldButton/TapAndHoldButton.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      </S.Container>
+    );
+  }
+
+  useEffect(() => {
+    if (onCompletedState) {
+      Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
+    }
```
Comment (rachelyellow) [inline src/components/TapAndHoldButton/TapAndHoldButton.tsx:47]: if (onCompletedState) Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);

File: src/components/TapAndHoldButton/index.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  const onComplete = () => {
+    alert("Complete");
+  };
+  return (
+    <TapAndHoldButton
+      onCompletePress={onComplete}
+      initialText="Tap and Hold to Confirm Delivery"
+      completedText="Clock Out"
+    />
+  );
```
Comment (rachelyellow) [inline src/components/TapAndHoldButton/index.stories.tsx:15]: <TapAndHoldButton
      onCompletePress={() => alert("Complete")}
      initialText="Tap and Hold to Confirm Delivery"
      completedText="Clock Out"
    />

**Review by rachelyellow — APPROVED** [review]

File: src/components/TapAndHoldButton/TapAndHoldButton.tsx (hunk 1 above)
Comment (normancapule) [reply src/components/TapAndHoldButton/TapAndHoldButton.tsx]: 👍

File: src/components/TapAndHoldButton/TapAndHoldButton.tsx (hunk 2 above)
Comment (normancapule) [reply src/components/TapAndHoldButton/TapAndHoldButton.tsx]: 👍

File: src/components/TapAndHoldButton/index.stories.tsx (hunk 3 above)
Comment (normancapule) [reply src/components/TapAndHoldButton/index.stories.tsx:15]: got it! thanks :)

**Pull request merged**